### ISSUE TITLE
Travis does not build NEST with Python

### ISSUE THREAD

**Issue by heplesser** [nest/nest-simulator#1593]
With the merge of #1576, specifically commit 5bef80375d95d7af52ad95ac401f69f70ef8c6d9 (see Travis builds [4523.2](https://travis-ci.org/github/nest/nest-simulator/jobs/683182625) vs [4522.2](https://travis-ci.org/github/nest/nest-simulator/jobs/683178076). The problem seems to be that since that commit, we have mix-up of Python executable and libraries:

```
-- Found PythonInterp: /home/travis/virtualenv/python3.6.10/bin/python3 (found suitable version "3.6.10", minimum required is "3") 
1573-- Could NOT find PythonLibs: Found unsuitable version "3.6.9", but required is exact version "3.6.10" (found /usr/lib/x86_64-linux-gnu/libpython3.6m.so)
1574
```

Beyond fixing this acute problem, this also is worrisome because the testsuite still reports success on the test configurations with Python even if NEST is built without Python. Can we add a test that checks that our test configuration is actually honoured? Should Cmake simply fail when Python is requested but not found?

**Comment by lekshmideepu** [comment]
Yes, the problem is critical. I figured out that there is a hard coded path in the build script causing the issue. I could imagine that the Cmake fails when Python libraries are not found for some reason although it is requested.

**Comment by heplesser** [comment]
Fixed by #1596.
